### PR TITLE
Add C implementation for `@stdlib/number/float32/base/exponent`

### PR DESCRIPTION
**Pull request by Pranavchiku**
## Checklist

- [x] update readme.md
- [ ] include/stdlib/number/float32/base
- [ ] src
- [ ] include.gypi
- [ ] manifest.json
- [ ] binding.gyp
- [ ] examples
- [ ] benchmark
- [ ] test
* * *

@kgryte if you could please assign stdlib bot.

### CODE REVIEW

**Comment by kgryte** [comment]
Done.

**Comment by kgryte** [comment]
Heads up that we are running some jobs in the workflow queue, so I am not sure how soon the bot will get around to scaffolding the package.

In the meantime, feel free to create PRs with the READMEs of other packages missing native implemenations.

File: lib/node_modules/@stdlib/number/float32/base/exponent/README.md [hunk 1]
```diff
@@ -91,6 +91,94 @@ for ( i = 0; i < 100; i++ ) {
...
+### Examples
+
+```c
+#include "stdlib/number/float32/base/exponent.h"
+#include <stdint.h>
+#include <stdio.h>
+#include <inttypes.h>
+
+int main() {
+    double x[] = { 4.0, 0.0, -0.0, 1.0, -1.0, 3.14, -3.14, 1.0e308, -1.0e308, 1.0/0.0, -1.0/0.0, 0.0/0.0 };
```
Comment (kgryte) [inline lib/node_modules/@stdlib/number/float32/base/exponent/README.md]: These should be floats.

File: lib/node_modules/@stdlib/number/float32/base/exponent/README.md [hunk 2]
```diff
@@ -91,6 +91,94 @@ for ( i = 0; i < 100; i++ ) {
...
+
+    int32_t out;
+    int i;
+    for ( i = 0; i < 12; i++ ) {
+        out = stdlib_base_float32_exponent( x[ i ] );
+        printf( "%lf => out: %" PRId32 "\n", x[ i ], out );
```
Comment (kgryte) [inline lib/node_modules/@stdlib/number/float32/base/exponent/README.md]: This should be `%f`.

File: lib/node_modules/@stdlib/number/float32/base/exponent/README.md [hunk 3]
```diff
@@ -91,6 +91,94 @@ for ( i = 0; i < 100; i++ ) {
...
+```
+
+#### stdlib_base_float32_exponent( x )
+
+Returns an integer corresponding to the unbiased exponent of a [single-precision floating-point number][ieee754].
+
+```c
+#include <stdint.h>
+
+int32_t out = stdlib_base_float32_exponent( 3.14 );
```
Comment (kgryte) [inline lib/node_modules/@stdlib/number/float32/base/exponent/README.md]:
```suggestion
int32_t out = stdlib_base_float32_exponent( 3.14f );
```

File: lib/node_modules/@stdlib/number/float32/base/exponent/README.md [hunk 4]
```diff
@@ -91,6 +91,94 @@ for ( i = 0; i < 100; i++ ) {
...
+```
+
+The function accepts the following arguments:
+
+-   **x**: `[in] double` input value.
```
Comment (kgryte) [inline lib/node_modules/@stdlib/number/float32/base/exponent/README.md]:
```suggestion
-   **x**: `[in] float` input value.
```

File: lib/node_modules/@stdlib/number/float32/base/exponent/README.md [hunk 5]
```diff
@@ -91,6 +91,94 @@ for ( i = 0; i < 100; i++ ) {
...
+
+```c
+int32_t stdlib_base_float32_exponent( const double x );
```
Comment (kgryte) [inline lib/node_modules/@stdlib/number/float32/base/exponent/README.md]:
```suggestion
int32_t stdlib_base_float32_exponent( const float x );
```

File: lib/node_modules/@stdlib/number/float32/base/exponent/README.md (hunk 1 above)
Comment (kgryte) [inline lib/node_modules/@stdlib/number/float32/base/exponent/README.md]:
```suggestion
    float x[] = { 4.0f, 0.0f, -0.0f, 1.0f, -1.0f, 3.14f, -3.14f, 1.0e38f, -1.0e38f, 1.0f/0.0f, -1.0f/0.0f, 0.0f/0.0f };
```

File: lib/node_modules/@stdlib/number/float32/base/exponent/README.md (hunk 2 above)
Comment (kgryte) [inline lib/node_modules/@stdlib/number/float32/base/exponent/README.md]:
```suggestion
        printf( "%f => out: %" PRId32 "\n", x[ i ], out );
```

**Comment by kgryte** [comment]
@Pranavchiku The scaffold ran; however, after looking through the generated content, a number of things are subtly incorrect. If you use the scaffolded C content, you'll want to go through things with a careful attention to detail.

Generating C content is still a WIP, and we'll continue iterating to hopefully improve what it's able to generate.

**Comment by Pranavchiku** [comment]
Yeah sure! I will look into those and finish up this PR asap.

**Comment by Pranavchiku** [comment]
Okay so, it gives me following logs
```
(base) pranavchiku@pranavchiku-Yoga-Slim-7-14IIL05:~/stdlib(exponent_float32)$ git merge origin/scaffold-609/number/float32/base/exponent
merge: origin/scaffold-609/number/float32/base/exponent - not something we can merge
```

where `origin` points to 
```
origin  https://github.com/stdlib-js/stdlib.git (fetch)
origin  https://github.com/stdlib-js/stdlib.git (push)
```

**Comment by Pranavchiku** [comment]
Apologies for keeping this PR on hold for long.

I actually messed up this PR on trying to merge the files, and hence created separate PR #611.

**Pull request closed without merging**